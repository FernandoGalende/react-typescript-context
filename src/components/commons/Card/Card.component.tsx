import React, {Fragment} from 'react';
import {City} from '../../../interfaces/app.interfaces';

import './Card.scss';

type CardProps = {
	city: City;
};

const Card: React.FC<CardProps> = ({city}) => (
	<Fragment>
		<div className='container'>
			<div className='image-block' />
			<input type='checkbox' />
			<div className='text-block'>
				<h4>{city.name}</h4>
				<h5>{city.chineseName}</h5>
			</div>
		</div>
	</Fragment>
);

export default Card;
