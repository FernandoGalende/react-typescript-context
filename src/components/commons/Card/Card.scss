@import '../../../styles/partials';
@import "../../../styles/variables";
@import "../../../styles/guide-utils";

.card-container {
	box-sizing: border-box;
	padding-left: 30px;
	height: 4em;
	display: flex;
	align-items: center;
	transition: background-color 300ms;
	@include mobile {
		height: 3em;
		padding-left: 8px;
	}
	input[type=checkbox] {
		position: relative;
		cursor: pointer;
		margin-left: 30px;
		@include mobile {
			margin-left: 8px;
		}
	}
	input[type=checkbox]:before {
		content: "";
		display: block;
		position: absolute;
		width: 16px;
		height: 16px;
		top: 0;
		left: 0;
		border: 2px solid #555555;
		border-radius: 3px;
		background-color: $White;
		@include mobile {
			width: 10px;
			height: 10px;
		}
	}
	input[type=checkbox]:checked:after {
		content: "";
		display: block;
		width: 5px;
		height: 10px;
		border: solid black;
		border-width: 0 2px 2px 0;
		-webkit-transform: rotate(45deg);
		-ms-transform: rotate(45deg);
		transform: rotate(45deg);
		position: absolute;
		top: 2px;
		left: 6px;
		@include mobile {
			width: 2.5px;
			height: 5px;
			top: 2px;
			left: 4.5px;
		}
	}
	.text-block {
		margin-left: 30px;
		h5 {
			line-height: normal;
			@extend %text-h4;
		}
		h6 {
			line-height: normal;
			@extend %text-note;
		}
		@include mobile {
			margin-left: 8px;
		}
	}
	.image-block {
		border-radius: 50%;
		height: 50px;
		width: 50px;
		background-color: $Grey10;
		@include mobile {
			width: 20px;
			height: 20px;
		}
	}
	.show-image {
		background-image: url("../../../assents/china.svg");
		background-repeat: no-repeat;
		background-size: cover;
		background-position: center;
		box-sizing: border-box;
	}
}

.container:hover {
	background-color: $Grey10;
}
