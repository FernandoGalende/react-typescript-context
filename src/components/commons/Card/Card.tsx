import React, {Fragment} from 'react';
import {City} from '../../../store/types';

import './Card.scss';

type CardProps = {
	city: City;
	onSelect: Function;
};

export const Card: React.FC<CardProps> = ({city, onSelect}) => {
	const handleChange = () => {
		city.selected = !city.selected;
		onSelect(city);
	};

	return (
		<Fragment>
			<div className='card-container' data-testid='card-test'>
				<div className='image-block' data-testid='image-test' />
				<input type='checkbox' onChange={handleChange} checked={city.selected} data-testid='input-test' />
				<div className='text-block'>
					<h5 data-testid='name-test'>{city.name}</h5>
					<h6 data-testid='chineseName-test'>{city.chineseName}</h6>
				</div>
			</div>
		</Fragment>
	);
};
