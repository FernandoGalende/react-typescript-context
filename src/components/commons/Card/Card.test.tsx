import React from 'react';
import {render, fireEvent} from '@testing-library/react';
import {Card} from './Card';

const mockFuntion = jest.fn();
const mockCity = {
	id: 'fakeId',
	name: 'fakeName',
	chineseName: 'fackeChineseName',
	selected: false
};

describe('<SearchBar />', () => {
	let getByTestId: Function;
	let container: HTMLElement;

	beforeEach(() => {
		const queries = render(
			<Card onSelect={mockFuntion} city={mockCity}>
				click!
			</Card>
		);
		container = queries.container;
		getByTestId = queries.getByTestId;
	});

	it('match snapshot', () => {
		expect(container).toMatchSnapshot();
	});

	it('should input exist', () => {
		expect(getByTestId('card-test')).toBeInTheDocument();
	});

	it('should has properly name', () => {
		expect(getByTestId('name-test')).toHaveTextContent('fakeName');
	});

	it('should has properly name', () => {
		expect(getByTestId('chineseName-test')).toHaveTextContent('fackeChineseName');
	});

	it('input should fire output function', async () => {
		const input = getByTestId('input-test');
		fireEvent.click(input);
		expect(mockFuntion).toHaveBeenCalled();
	});

	it('text on input should have properly test', async () => {
		const input = getByTestId('input-test');
		fireEvent.change(input, {target: {checked: true}});
		expect(mockFuntion).toHaveBeenCalledTimes(1);
	});
});
