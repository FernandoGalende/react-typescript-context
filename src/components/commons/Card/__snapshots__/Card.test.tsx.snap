// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar /> match snapshot 1`] = `
<div>
  <div
    class="card-container"
    data-testid="card-test"
  >
    <div
      class="image-block"
      data-testid="image-test"
    />
    <input
      data-testid="input-test"
      type="checkbox"
    />
    <div
      class="text-block"
    >
      <h5
        data-testid="name-test"
      >
        fakeName
      </h5>
      <h6
        data-testid="chineseName-test"
      >
        fackeChineseName
      </h6>
    </div>
  </div>
</div>
`;
